src/transaction-runner.coffee

```diff
@@ -455,7 +455,9 @@ class TransactionRunner
     for key, value of transaction.request.headers
       caseInsensitiveRequestHeadersMap[key.toLowerCase()] = key
 
-    if not caseInsensitiveRequestHeadersMap['content-length'] and transaction.request['body'] != ''
+    if not caseInsensitiveRequestHeadersMap['content-length'] and
+        transaction.request['body']? and
+        transaction.request['body'] != ''
```

Comment:
I think we should be all right even with this:

```coffee
if transaction.request.body and not caseInsensitiveRequestHeadersMap['content-length']
  ...
```

Also, I'm thinking whether something like this wouldn't make sense as well:

```coffee
if not caseInsensitiveRequestHeadersMap['content-length']
  if transaction.request.body
    logger.verbose('Calculating Content-Length of the request body')
    ...
  else
    transaction.request.headers['Content-Length'] = 0
```

What do you think?